src/radix_connect/wallet_interaction/dapp_wallet_interaction/dapp_to_wallet/interaction_items/transaction/unvalidated_transaction_manifest.rs

```diff
@@ -0,0 +1,98 @@
+use crate::prelude::*;
+
+#[derive(
+    Clone, PartialEq, Eq, Serialize, Deserialize, Debug, uniffi::Record,
+)]
+pub struct UnvalidatedTransactionManifest {
+    #[serde(rename = "transactionManifest")]
+    pub transaction_manifest_string: String,
+
+    pub blobs: Blobs,
+}
+
+impl UnvalidatedTransactionManifest {
+    pub fn new(
+        transaction_manifest_string: impl AsRef<str>,
+        blobs: impl Into<Blobs>,
+    ) -> Self {
+        Self {
+            transaction_manifest_string: transaction_manifest_string
+                .as_ref()
+                .to_owned(),
+            blobs: blobs.into(),
+        }
+    }
+}
+
+impl From<TransactionManifest> for UnvalidatedTransactionManifest {
+    fn from(transaction_manifest: TransactionManifest) -> Self {
+        Self {
+            transaction_manifest_string: transaction_manifest
+                .instructions_string(),
+            blobs: transaction_manifest.blobs().clone(),
+        }
+    }
+}
+
+impl UnvalidatedTransactionManifest {
+    pub fn transaction_manifest(
+        &self,
+        network_id: NetworkID,
+    ) -> Result<TransactionManifest> {
+        TransactionManifest::new(
+            self.transaction_manifest_string.clone(),
+            network_id,
+            self.blobs.clone(),
+        )
+    }
+}
+
+impl HasSampleValues for UnvalidatedTransactionManifest {
+    fn sample() -> Self {
+        Self::new(
+            TransactionManifest::sample().instructions_string(),
+            Blobs::default(),
+        )
+    }
+
+    fn sample_other() -> Self {
+        Self::new(
+            TransactionManifest::sample_other().instructions_string(),
+            Blobs::default(),
```

Comment:
maybe good if one of the sample values had blobs? so maybe use `Blobs::sample_other()` here.